msticpy/vis/entity_graph_tools.py

```diff
@@ -0,0 +1,559 @@
+                hide=True,
+                width=800,
+            )
+        elif tl_type == "discreet":
+            timeline = display_timeline(
+                tl_df.dropna(subset=["TimeGenerated"]),
+                group_by="Type",
+                title="Entity Timeline",
+                time_column="TimeGenerated",
+                source_columns=["Name", "Description", "Type", "TimeGenerated"],
+                hide=True,
+                width=800,
+            )
+        plot_layout = column(graph, timeline) if timeline else graph
+        if not hide:
+            show(plot_layout)
+        return plot_layout
+
+    def add_entity(self, ent: Entity, attached_to: str = None):
+        """
+        Add an entity to the graph.
+
+        Parameters
+        ----------
+        ent : Entity
+            The entity object to add the graph
+        attached_to : str, optional
+            The name of the node to attach the entity to, by default None
+
+        """
+        self._add_entity_node(ent, attached_to)
+
+    def add_incident(self, incident: Union[Incident, Alert, pd.DataFrame]):
+        """
+        Add another incident or set of incidents to the graph.
+
+        Parameters
+        ----------
+        incident : Union[Incident, Alert, pd.DataFrame]
+            This can be an alert, and incident or a DataFrame of alerts or incidents
+
+        """
+        inc = None
+        if isinstance(incident, pd.DataFrame):
+            for row in incident.iterrows():
+                if "name" in row[1]:
+                    inc = Incident(src_event=row[1])
+                elif "AlertName" in row[1]:
+                    inc = Alert(src_event=row[1])  # type: ignore
+                self._add_incident_or_alert_node(inc)
+        else:
+            self._add_incident_or_alert_node(incident)
+
+    def add_note(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        attached_to: Union[str, List] = None,
+    ):
+        """
+        Add a node to the graph representing a note or comment.
+
+        Parameters
+        ----------
+        name : str
+            The name of the node to add
+        description : Optional[str], optional
+            A description of the note, by default None
+        attached_to : Union[str, List], optional
+            What existing nodes on the graph to attach it the note to, by default None
+        user: str, optional
+            What user to associate the note with
+
+        """
+        self.alertentity_graph.add_node(
+            name,
+            Name=name,
+            Description=description,
+            Type="analystnote",
+            TimeGenerated=datetime.now(),
+        )
+        if attached_to:
+            if isinstance(attached_to, str):
+                attached_to = [attached_to]
+            for link in attached_to:
+                self.add_link(name, link)
+
+    def add_link(self, source: str, target: str):
```

Comment:
Possibly not show-stopping but the usual terminology for this is add_edge, delete (or remove)_edge.
I think we have this (at least add_edge in the entities.entity_graph module).